src/Core.zig

```diff
@@ -102,31 +135,19 @@ input_state: InputState,
 oom: std.Thread.ResetEvent = .{},
 
 pub fn init(core: *Core) !void {
-    // TODO: this needs to be removed.
-    const options: InitOptions = .{
-        .allocator = std.heap.c_allocator,
-    };
-    const allocator = options.allocator;
+    const allocator = std.heap.c_allocator;
 
     // TODO: fix all leaks and use options.allocator
     try mach.sysgpu.Impl.init(allocator, .{});
 
-    const main_window = try core.windows.new(.{
-        .title = options.title, // TODO
-        .framebuffer_format = undefined, // TODO: null?
-        .framebuffer_width = undefined, // TODO: null?
-        .framebuffer_height = undefined, // TODO: null?
-        .width = 1920 / 2,
-        .height = 1080 / 2,
-        .fullscreen = false,
-    });
-
-    // Copy window title into owned buffer.
-    var title: [256:0]u8 = undefined;
-    if (options.title.len < title.len) {
-        @memcpy(title[0..options.title.len], options.title);
-        title[options.title.len] = 0;
-    }
+    const main_window = try core.windows.new(.{});
```

Comment:
After this change, we should consider whether removing 'main window' as a concept might make sense - it might make the API (and multiple window support) better in general maybe.

Reply:
This could maybe address the '`try Platform.tick(core);` is called inside init' question you had.

Short-term, I think this is totally fine as-is though.